lib/httpheaders.h

```diff
@@ -0,0 +1,79 @@
+#ifndef HTTPHEADERS_H
+#define HTTPHEADERS_H
+
+#include <QtCore>
+#include "export.h"
+
+#define HTTP_METHOD(V) V(OPTIONS) \
+      V(GET) \
+      V(HEAD ) \
+      V(POST) \
+      V(PUT) \
+      V(DELETE) \
+      V(TRACE) \
+      V(CONNECT)
+
+namespace QSS {
+
+class QSS_EXPORT HttpHeaders final : QObject {
+
+    Q_OBJECT
+
+public:
+
+#define XX(NAME) NAME,
```

Comment:
No macro == better readability, less error prone. Also, no need to be defined in header file